heron/uploaders/src/java/com/twitter/heron/uploader/hdfs/HdfsUploader.java

```diff
@@ -65,33 +66,32 @@ protected boolean isLocalFileExists(String file) {
   }
 
   @Override
-  public URI uploadPackage() {
+  public URI uploadPackage() throws UploaderException {
     // first, check if the topology package exists
     if (!isLocalFileExists(topologyPackageLocation)) {
-      LOG.info("Topology file " + topologyPackageLocation + " does not exist.");
-      return null;
+      throw new UploaderException(
+        String.format("Topology file %s does not exist", topologyPackageLocation));
     }
 
     // if the dest directory does not exist, create it.
     if (!controller.exists(destTopologyDirectoryURI)) {
       LOG.info("The destination directory does not exist; creating it.");
       if (!controller.mkdirs(destTopologyDirectoryURI)) {
-        LOG.severe("Failed to create directory: " + destTopologyDirectoryURI);
-        return null;
+        throw new UploaderException(
+            String.format("Failed to create directory: %s", destTopologyDirectoryURI));
```

Comment:
more context for the user here (i.e., why are we trying to create this directory?)